plonky2x/core/src/frontend/hash/blake2/curta.rs

```diff
@@ -227,7 +227,26 @@ impl<L: PlonkParameters<D>, const D: usize> CircuitBuilder<L, D> {
         input_byte_length: U32Variable,
     ) -> U32Variable {
         let chunk_size = self.constant::<U32Variable>(128);
-        self.div(input_byte_length, chunk_size)
+        let last_chunk_idx = self.div(input_byte_length, chunk_size);
+
+        // Check to see if the input length is a multiple of 128 and it's not zero length. If it is,
+        // we need to subtract 1 from the last chunk index.
+
+        // Find out if the input length is a multiple of 128.
+        let mut tmp = self.mul(chunk_size, last_chunk_idx);
+        tmp = self.sub(tmp, input_byte_length);
+        let is_multiple = self.is_zero(tmp.variable);
+
+        // Find out if the input length is not zero.
+        let is_zero_length = self.is_zero(input_byte_length.variable);
+        let is_not_zero_length = self.not(is_zero_length);
+
+        // Find out if we need to subtract 1 from the last chunk index.
+        let do_minus_one = self.and(is_multiple, is_not_zero_length);
+        let one = self.one();
+        let last_chunk_idx_minus_one = self.sub(last_chunk_idx, one);
+
+        self.select(do_minus_one, last_chunk_idx_minus_one, last_chunk_idx)
```

Comment:
```
if input_byte_length == 0 {
      0
  } else {
      (input_byte_length - 1) / 128
  }
```

This is cleaner, doesn't require `mul` logic.

Reply:
How strongly do you feel about this?  I actually think that my implementation is more easier to understand.  E.g. the (input_byte_length - 1)/128 kind of implies the logic that I'm doing.

Reply:
The comments above the logic should be more clear to specify what's going on then.

I do think the following is more clear to follow: "If the message length is 0, the last idx is 0. If it is non-zero, the message length is `input_byte_length - 1` / 128. Don't need to add in the additional "find out if input length is a multiple of 128"

Reply:
@kevjue Please clean up the comments here ^

Reply:
This is personal pref, but I also try to avoid "we need to" when writing comments. Ex. "If it is ~~we need to~~ subtract 1 from the last chunk index."

Reply:
Fixed